tcl/org.testeditor.tcl.dsl.ui/src/org/testeditor/tcl/dsl/ui/editor/DropTargetXtextEditorListener.xtend

```diff
@@ -0,0 +1,311 @@
+package org.testeditor.tcl.dsl.ui.editor
+
+import java.util.List
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.util.EcoreUtil
+import org.eclipse.swt.dnd.DropTargetAdapter
+import org.eclipse.swt.dnd.DropTargetEvent
+import org.eclipse.xtext.resource.XtextResource
+import org.eclipse.xtext.util.concurrent.IUnitOfWork
+import org.testeditor.aml.ComponentElement
+import org.testeditor.aml.InteractionType
+import org.testeditor.aml.TemplateText
+import org.testeditor.aml.TemplateVariable
+import org.testeditor.tcl.AbstractTestStep
+import org.testeditor.tcl.ComponentTestStepContext
+import org.testeditor.tcl.SpecificationStepImplementation
+import org.testeditor.tcl.StepContentElement
+import org.testeditor.tcl.TclModel
+import org.testeditor.tcl.TestCase
+import org.testeditor.tcl.TestStep
+import org.testeditor.tsl.StepContentText
+import org.testeditor.tsl.StepContentVariable
+import org.testeditor.aml.Component
+import org.eclipse.xtext.EcoreUtil2
+import org.eclipse.jface.text.source.SourceViewer
+import org.eclipse.jface.viewers.ISelection
+import org.eclipse.jface.util.LocalSelectionTransfer
+import org.eclipse.jface.viewers.TreeSelection
+import org.eclipse.jface.text.source.ISourceViewer
+import org.testeditor.tcl.impl.TclFactoryImpl
+import org.testeditor.tsl.impl.TslFactoryImpl
+import javax.inject.Inject
+import org.eclipse.swt.dnd.DND
+import org.eclipse.jface.text.TextSelection
+import org.eclipse.xtext.resource.ILocationInFileProvider
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.testeditor.tsl.impl.StepContentTextImpl
+import org.testeditor.tcl.impl.SpecificationStepImplementationImpl
+import org.eclipse.xtext.ui.editor.contentassist.ContentAssistContext
+import org.eclipse.xtext.util.StringInputStream
+import java.util.Collections
+import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.xtext.resource.XtextResource
+import java.net.URI
+
+class DropTargetXtextEditorListener extends DropTargetAdapter {
+
+	@Inject protected TclFactoryImpl tclFactory
+	@Inject protected TslFactoryImpl tslFactory
+
+	@Inject extension ILocationInFileProvider
+	@Inject protected ContentAssistContext.Factory contentAssistFactory
+	
+	@Accessors(PUBLIC_SETTER)
+	private DropTargetXtextEditor editor
+
+	override void dragEnter(DropTargetEvent event) {
+		if ("org.testeditor.tcl.dsl.Tcl" != editor.languageName) {
```

Comment:
That's true but on the other hand: what if the name of the language is changed? The person doing that will probably not think of this part of the code. We could probably name the variable `tclLanguageName` to make it more clear.

Apart from that: if the listener is only registered on the Tcl editor, can this ever be not true?